tests/test_tools.py

```diff
@@ -55,6 +57,32 @@ async def invalid_tool(ctx: RunContext[None]) -> str:  # pragma: no cover
     )
 
 
+def test_builtin_tool_registry():
+    """
+    Test that built-in functions can be registered as tools.
+    """
+
+    with pytest.raises(UserError, match='Error generating schema for'):
+        Agent(TestModel(), tools=[max])
+
+    with pytest.raises(UserError, match='Error generating schema for'):
+        agent = Agent(TestModel())
+        agent.tool_plain(min)
+
+    def test_tool(*args):  # type: ignore[reportUnknownParameterType, reportUnknownArgumentType]
+        pass
+
+    with pytest.raises(ValueError, match='no signature found for'):
+        signature(max)
+    sig = signature(test_tool)  # type: ignore[reportUnknownArgumentType]
+    type_hints = get_function_type_hints(test_tool)  # type: ignore[reportUnknownArgumentType]
+    first_param_name = next(iter(sig.parameters.keys()))
+    with pytest.raises(KeyError):
+        type_hints[first_param_name]  # KeyError
+
+    assert first_param_name not in type_hints
```

Comment:
I'm not sure what the goal of this part of the test is, as it doesn't test verify actual Pydantic AI functionality, just "look this doesn't work". I think we can remove this, because the point of this PR is just the new `Error getting signature for` error we already test above, right?